extensions/amp-a4a/0.1/test/test-a4a-integration.js

```diff
@@ -0,0 +1,221 @@
+import {adConfig} from '../../../../ads/_config';
+import {a4aRegistry} from '../../../../ads/_a4a-config';
+import {
+    resetScheduledElementForTesting,
+    upgradeOrRegisterElement,
+} from '../../../../src/custom-element';
+import * as sinon from 'sinon';
+
+chai.Assertion.addMethod('renderedInFriendlyIframe', function(srcdoc) {
+  const obj = this._obj;
+  this.assert(obj,
+      'amp ad element should be #{exp}, got #{act}',  // if true message
+      'amp ad element should not be #{exp}, got #{act}',  // if negated message
+      'truthy',  // expected
+      obj);      // actual
+  const elementName = obj.tagName.toLowerCase();
+  const child = obj.querySelector('iframe[srcdoc]');
+  this.assert(child,
+      `child of amp ad element ${elementName} should be #{exp}, got #{act}`,
+      `child of amp ad element ${elementName} should not be #{exp}, got #{act}`,
+      'truthy',
+      child);
+  this.assert(child.getAttribute('srcdoc').indexOf(srcdoc) >= 0,
+      `iframe child of amp ad element ${elementName} should contain #{exp}, ` +
+      'was #{act}',
+      `iframe child of amp ad element ${elementName} should not contain ` +
+      '#{exp}, was #{act}',
+      srcdoc,
+      child.getAttribute('srcdoc'));
+  const childBody = child.contentDocument.body;
+  this.assert(childBody,
+      `iframe child of amp ad element ${elementName} should have #{exp}`,
+      `iframe child of amp ad element ${elementName} should have #{exp}, ` +
+      'got #{act}',
+      'body tag',
+      childBody);
+  [obj, child, childBody].forEach(toTest => {
+    expect(toTest).to.be.visible;
+  });
+});
+
+chai.Assertion.addMethod('renderedInXDomainIframe', function(src) {
+  const obj = this._obj;
+  this.assert(obj,
+      'amp ad element should be #{exp}, got #{act}',  // if true message
+      'amp ad element should not be #{exp}, got #{act}',  // if negated message
+      'truthy',  // expected
+      obj);      // actual
+  const elementName = obj.tagName.toLowerCase();
+  const friendlyChild = obj.querySelector('iframe[srcdoc]');
+  this.assert(!friendlyChild,
+      `child of amp ad element ${elementName} should not be cross-domain`,
+      `child of amp ad element ${elementName} should be cross-domain`);
+  const child = obj.querySelector('iframe[src]');
+  this.assert(child,
+      `child of amp ad element ${elementName} should be #{exp}, got #{act}`,
+      `child of amp ad element ${elementName} should not be #{exp}, got #{act}`,
+      'truthy',
+      child);
+  this.assert(child.getAttribute('src').indexOf(src) >= 0,
+      `iframe child of amp ad element ${elementName} src should contain ` +
+      '#{exp}, was #{act}',
+      `iframe child of amp ad element ${elementName} src should not contain ` +
+      '#{exp}, was #{act}',
+      src,
+      child.getAttribute('src'));
+  [obj, child].forEach(toTest => {
+    expect(toTest).to.be.visible;
+  });
+});
+
+describe('integration test: a4a', () => {
```

Comment:
I'm fine to leave it as is. In general, `addElement` is OK for a "integration" test like this one, but we'd like to avoid for unit test (no need to initialize the AMP runtime).